arkfilepicker/src/main/java/space/taran/arkfilepicker/folders/FoldersRepo.kt

```diff
@@ -119,11 +113,57 @@ class FoldersRepo(private val appCtx: Context) {
         } ?: listOf(fav)
         folders = mutFolders
 
-        val arkFolder = root.arkFolder()
-        require(arkFolder.exists()) { "Ark folder must exist" }
-        val favsFile = arkFolder.arkFavorites()
-        val jsonFavs = JsonFavorites(folders[root]!!.map(Path::toString))
-        favsFile.writeText(klaxon.toJsonString(jsonFavs))
+        writeFavorites(root)
+    }
+
+    suspend fun forgetRoot(root: Path) =
+        withContext(Dispatchers.IO) {
+            if (folders.containsKey(root)) {
+                val mFolders = folders.toMutableMap()
+                mFolders.remove(root)
+                folders = mFolders
+                writeRoots()
+            }
+        }
+
+    suspend fun deleteRoot(root: Path) =
+        withContext(Dispatchers.IO) {
+            if (folders.containsKey(root)) {
+                forgetRoot(root)
+                deleteFilesRecursively(root)
+            }
+        }
+
+    suspend fun forgetFavorite(root: Path, favorite: Path) =
+        withContext(Dispatchers.IO) {
+            if (folders.containsKey(root)) {
+                val favorites = folders[root]
+                val favRelative = root.relativize(favorite)
+                if (favorites != null && favorites.contains(favRelative)) {
+                    val mFavorites = favorites.toMutableList()
```

Comment:
Do we really need to convert to mutable structure here?

Reply:
Am not sure, but I guess immutable deletion can work too. Like your comment above :)